sql/spanner/inspect.go

```diff
@@ -0,0 +1,738 @@
+		if err := rows.Scan(&tableName, &checkName, &clause, &spannerState); err != nil {
+			return fmt.Errorf("spanner: scanning check: %w", err)
+		}
+		t, ok := s.Table(tableName)
+		if !ok {
+			return fmt.Errorf("table %q was not found in schema", tableName)
+		}
+		t.AddChecks(&schema.Check{
+			Name: checkName,
+			Expr: clause,
+		})
+	}
+	return nil
+}
+
+// schemas returns the list of the schemas in the database.
+func (i *inspect) schemas(ctx context.Context, opts *schema.InspectRealmOption) ([]*schema.Schema, error) {
+	var (
+		args    []any
+		sArgs   []string
+		query   = schemasQuery
+		schemas []*schema.Schema
+	)
+	if opts != nil {
+		switch n := len(opts.Schemas); {
+		case n == 0:
+			query = schemasQuery
+		case n > 0:
+			query = schemasQueryArgs
+			for _, s := range opts.Schemas {
+				if s == defaultSchemaNameAlias {
+					s = ""
+				}
+				sArgs = append(sArgs, s)
+			}
+			args = append(args, sArgs)
+		}
+	}
+	rows, err := i.QueryContext(ctx, query, args...)
+	if err != nil {
+		return nil, fmt.Errorf("spanner: querying schemas: %w", err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var name string
+		if err := rows.Scan(&name); err != nil {
+			return nil, err
+		}
+		if name == "" {
+			name = defaultSchemaNameAlias
+		}
+		schemas = append(schemas, &schema.Schema{
+			Name: name,
+		})
+	}
+	if err := rows.Close(); err != nil {
+		return nil, err
+	}
+	return schemas, nil
+}
+
+func (i *inspect) querySchema(ctx context.Context, query string, s *schema.Schema) (*sql.Rows, error) {
+	tables := []string{}
+	for _, t := range s.Tables {
+		tables = append(tables, t.Name)
+	}
+	sName := s.Name
+	if sName == defaultSchemaNameAlias {
+		sName = ""
+	}
+	return i.QueryContext(ctx, query, sName, tables)
+}
+
+func defaultExpr(c *schema.Column, x string) schema.Expr {
+	switch {
+	case sqlx.IsLiteralBool(x), sqlx.IsLiteralNumber(x), sqlx.IsQuoted(x, '\''):
+		return &schema.Literal{V: x}
+	default:
+		// Try casting or fallback to raw expressions (e.g. column text[] has the default of '{}':text[]).
+		if v, ok := canConvert(c.Type, x); ok {
+			return &schema.Literal{V: v}
+		}
+		return &schema.RawExpr{X: x}
+	}
+}
+
+func canConvert(t *schema.ColumnType, x string) (string, bool) {
+	r := t.Raw
+	i := strings.Index(x, "::"+r)
+	if i == -1 || !sqlx.IsQuoted(x[:i], '\'') {
+		return "", false
+	}
+	q := x[0:i]
+	x = x[1 : i-1]
+	switch t.Type.(type) {
+	case *schema.BoolType:
+		if sqlx.IsLiteralBool(x) {
+			return x, true
+		}
+	case *schema.DecimalType, *schema.IntegerType, *schema.FloatType:
+		if sqlx.IsLiteralNumber(x) {
+			return x, true
+		}
+	case *schema.BinaryType, *schema.JSONType, *schema.SpatialType, *schema.StringType, *schema.TimeType, *StringType, *BytesType:
+		return q, true
+	}
+	return "", false
+}
+
+func removeEmptyStrings(s []string) []string {
+	var r []string
+	for _, str := range s {
+		if str != "" {
+			r = append(r, str)
+		}
+	}
+	return r
```

Comment:
```suggestion
func removeEmptyStrings(ss []string) []string {
	var r []string
	for _, s := range ss {
		if s != "" {
			r = append(r, s)
		}
	}
	return r
```
No special reason, just a naming style.